pallets/vector/src/lib.rs

```diff
@@ -175,6 +188,27 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type WhitelistedDomains<T> = StorageValue<_, BoundedVec<u32, ConstU32<10_000>>, ValueQuery>;
 
+	/// The storage for the step function identifier and the rotate function identifier.
+	/// Step function id is used to distinguish step-related functionality within the fulfill_call function.
+	/// Rotate function id is used to handle rotate-related functionality within the fulfill_call function.
+	/// When the provided function_id matches the step/rotate function identifier, specific logic related to step/rotate functions is executed.
+	/// The order of storage is (step_function_id, rotate_function_id)
+	#[pallet::storage]
+	#[pallet::getter(fn function_ids)]
+	pub type FunctionIds<T: Config> = StorageValue<_, Option<(H256, H256)>, ValueQuery>;
+
+	/// Step verification key storage
+	#[pallet::storage]
+	#[pallet::getter(fn step_verification_key)]
+	pub type StepVerificationKey<T: Config> =
+		StorageValue<_, Option<BoundedVec<u8, ConstU32<10_000>>>, ValueQuery>;
```

Comment:
What are our expectations on the size of Step Verification Keys? Is 10k enough to fit current and future keys?

Reply:
I took the same specification as stuff above. It was already a bounded vec somewhere. Right now it's roughly 3 to 4 times what's necessary. We also can update it or even use new storage and delete this one incase we don't want to do a migration for that haha.